### PR TITLE
Add Speller And MRT  Doc In GA

### PR DESCRIPTION
**Pull request by ZChristine**
https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/13006

### CODE REVIEW

File: api-reference/v1.0/resources/searchrequest.md [hunk 1]
```diff
@@ -22,17 +22,20 @@ The JSON blob contains the types of resources expected in the response, the unde
...
 |from|Int32|Specifies the offset for the search results. Offset 0 returns the very first result. Optional.|
 |query|[searchQuery](searchquery.md)|Contains the query terms. Required.|
+|resultTemplateOptions|[resultTemplateOption](resultTemplateOption.md) collection|Provides the search result templates options to render search results from connectors.|
 |size|Int32|The size of the page to be retrieved. Optional.|
 |sortProperties|[sortProperty](sortProperty.md) collection|Contains the ordered collection of fields and direction to sort results. There can be at most 5 sort properties in the collection. Optional.|
+|resultTemplateOptions|[resultTemplateOption](resultTemplateOption.md) collection|Provides the search result templates options for rendering connectors search results.|
+|queryAlterationOptions|[searchAlterationOptions](searchalterationoptions.md)|Query alteration options formatted in a JSON blob that contains two optional flags related to spelling correction. Optional. |
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchrequest.md:38]: For consistency, please move the property **queryAlterationOptions** to come after **query**.

File: api-reference/v1.0/resources/searchresponse.md [hunk 2]
```diff
@@ -19,6 +19,8 @@ Represents results from a search query, and the terms used for the query.
 |:-------------|:------------|:------------|
 |hitsContainers|[searchHitsContainer](searchhitscontainer.md) collection|A collection of search results.|
 |searchTerms|String collection|Contains the search terms sent in the initial search query.|
+|resultTemplates|[resultTemplate](resultTemplate.md) collection|A dictionary of resultTemplateIds and associated values, which include the name and JSON schema of the result templates.
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchresponse.md:22]: For consistency, **resultTemplate.md** must be in lower case format. Also, please apply bold formatting to resultTemplateIds.

File: api-reference/v1.0/resources/searchresponse.md [hunk 3]
```diff
@@ -19,6 +19,8 @@ Represents results from a search query, and the terms used for the query.
...
+|queryAlterationResponse|[alterationResponse](alterationResponse.md)|Provides details of query alteration response for spelling correction.|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchresponse.md:23]: For consistency with the alterationResponse topic, I recommend you to use "Provides information related to spelling corrections in the alteration response." as the description.

File: changelog/Microsoft.MicrosoftSearch.json [hunk 4]
```diff
@@ -395,6 +395,136 @@
       "CreatedDateTime": "2021-01-28T12:49:29.5221544Z",
       "WorkloadArea": "Search",
       "SubArea": "Analytics"
+    },
+    {
+      "ChangeList": [
+        {
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:401]: I can't find this change in your PR. This looks more like an issue in the changelog than a real addition. Please remove this entry from this changelog record.

File: changelog/Microsoft.MicrosoftSearch.json [hunk 5]
```diff
@@ -395,6 +395,136 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **searchAlterationType** enumeration type.",
+          "Target": "searchAlterationType"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Resource",
+          "ChangedApiName": "alterationResponse",
+          "ChangeType": "Addition",
+          "Description": "Added the [alterationResponse](https://docs.microsoft.com/en-us/graph/api/resources/alterationResponse?view=graph-rest-1.0) resource type.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:422]: GLOBAL: Please apply this change to _all_ **Description** properties in this changelog record.

- URLs must be provided in lower case format. In this case, please replace "alterationResponse" with "alterationresponse" => ...api/resources/alterationresponse?view...

File: changelog/Microsoft.MicrosoftSearch.json [hunk 6]
```diff
@@ -395,6 +395,136 @@
...
+          "ApiChange": "Resource",
+          "ChangedApiName": "resultTemplate",
+          "ChangeType": "Addition",
+          "Description": "Added the [resultTemplate](https://docs.microsoft.com/en-us/graph/api/resources/resultTemplate?view=graph-rest-1.0) resource type.",
+          "Target": "resultTemplate"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Resource",
+          "ChangedApiName": "resultTemplateDictionary",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:444]: This resource topic isn't part of this PR. Please, either add the **resultTemplateDictionary** resource topic to this PR or remove this entry.

File: changelog/Microsoft.MicrosoftSearch.json [hunk 7]
```diff
@@ -395,6 +395,136 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the [searchAlterationOptions](https://docs.microsoft.com/en-us/graph/api/resources/searchAlterationOptions?view=graph-rest-1.0) resource type.",
+          "Target": "searchAlterationOptions"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Property",
+          "ChangedApiName": "resultTemplateId",
+          "ChangeType": "Addition",
+          "Description": "Added the **resultTemplateId** property to [searchHit](https://docs.microsoft.com/en-us/graph/api/resources/searchHit?view=graph-rest-1.0) resource.",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:478]: Please, add "the" before 

- "[searchHit]" (line 478) => ...property to the [searchHit]...
- "[searchRequest]" (line 486)
- "[searchRequest]" (line 494)
- "[searchResponse]" (line 502)
- "[searchResponse]" (line 510)

File: changelog/Microsoft.MicrosoftSearch.json [hunk 8]
```diff
@@ -395,6 +395,136 @@
...
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
+          "ApiChange": "Property",
+          "ChangedApiName": "resultTemplates",
+          "ChangeType": "Addition",
+          "Description": "Added the **resultTemplates** property to [searchResponse](https://docs.microsoft.com/en-us/graph/api/resources/searchResponse?view=graph-rest-1.0) resource.",
+          "Target": "searchResponse"
+        },
+        {
+          "Id": "fca2af31-c9df-449e-b848-ecb564518c30",
```
Comment (JarbasHorst) [inline changelog/Microsoft.MicrosoftSearch.json:514]: This change looks more like an issue in the changelog than a real addition. Please remove this entry from this changelog record.

File: api-reference/v1.0/resources/resulttemplate.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  "optionalProperties": [
+
+  ],
+  "@odata.type": "microsoft.graph.resultTemplate",
+  "baseType": null
+}-->
+
+
+```json
+{
```
[the remark this answers is not in the document]
Comment (ZChristine) [reply api-reference/v1.0/resources/resulttemplate.md:39]: do you mean the json format from line 28 to line 35? this follow the format as other md file, for example resultinfo,md

File: changelog/Microsoft.MicrosoftSearch.json (hunk 4 above)
Comment (ZChristine) [reply changelog/Microsoft.MicrosoftSearch.json:401]: @JarbasHorst all the comment are resolved. can you check?

File: api-reference/v1.0/resources/searchresponse.md [hunk 10]
```diff
@@ -19,6 +19,8 @@ Represents results from a search query, and the terms used for the query.
 |:-------------|:------------|:------------|
```
Comment (JarbasHorst) [inline api-reference/v1.0/resources/searchresponse.md:19]: For consistency and to improve scanning, properties in the Properties table should be in alphabetical order. Please, alphabetize the list.

File: api-reference/v1.0/resources/search-api-overview.md [hunk 11]
```diff
@@ -130,7 +132,23 @@ Once the response is returned containing the collection of [searchBucket](search
...
 
-See [refine search results](/graph/search-concept-aggregation) for examples that show using aggregation to enhance and narrow down search results.
+For examples that show how to use aggregation to enhance and narrow down search results, see [Refine search results](/graph/search-concept-aggregation).
+
+## Request spelling correction
+
+Spelling correction is a popular way to handle mismatches between typos in a user query and the correct words in matched contents. When typos are detected in the original user query, you can get the search result either for the original user query or the corrected alternate query. You can also get the spelling correction information for typos in the **queryAlterationResponse** property of the [searchResponse](searchResponse.md).
+
+In the request body of the [query](/graph/api/search-query) method, specify the **queryAlterationOptions** that should be applied to the query for the spelling corrections. The description of **queryAlterationOptions** is defined in the [searchAlterationOptions](./searchalterationoptions.md).
```
[the remark this answers is not in the document]
Comment (ZChristine) [reply api-reference/v1.0/resources/search-api-overview.md:141]: update

File: api-reference/v1.0/resources/searchalterationoptions.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+Namespace: microsoft.graph
+
+Provides the search alteration options for spelling correction.
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|enableModification|Boolean|Indicates whether spelling modifications are enabled. If enabled, the user will get the search results for the corrected query *in case of no results* for the original query with typos. The [response](/graph/api/resources/searchresponse) will also include the spelling modification information in the **queryAlterationResponse** property. Optional.
+|enableSuggestion|Boolean|Indicates whether spelling suggestions are enabled. If enabled, the user will get the search results for the original search query and suggestions for spelling correction in the **queryAlterationResponse** property of the [response](/graph/api/resources/searchresponse) for the typos in the query. Optional.
```
[the remark this answers is not in the document]
Comment (ZChristine) [reply api-reference/v1.0/resources/searchalterationoptions.md:21]: update

File: api-reference/beta/resources/searchalterationoptions.md [hunk 13]
```diff
@@ -19,7 +19,7 @@ Provides the search alteration options for spelling correction.
 
 | Property     | Type        | Description |
 |:-------------|:------------|:------------|
-|enableSuggestion|Boolean|Indicates whether spelling suggestions are enabled. If enabled, user will get the search results for original search query and suggesting spelling correction in **queryAlterationResponse** property of the [response](/graph/api/resources/searchresponse?view=graph-rest-beta&preserve-view=true) for typos in query. Optional.|
+|enableSuggestion|Boolean|Indicates whether spelling suggestions are enabled. If enabled, user will get the search results for original search query and suggesting spelling correction in [queryAlterationResponse](/graph/api/resources/alterationresponse?view=graph-rest-beta&preserve-view=true) for typos in query. Optional.|
```
[the remark this answers is not in the document]
Comment (ZChristine) [reply api-reference/beta/resources/searchalterationoptions.md:22]: @JarbasHorst  the comment have been resolved. can we prioritize merging of this pr as we have timeline for publishing by the end of this week?

[the remark this answers is not in the document]
Comment (JarbasHorst) [reply api-reference/beta/resources/searchalterationoptions.md:22]: Hi @ZChristine, unfortunately, the comments **haven't** been resolved. The comments are marked as resolved but the feedback hasn't been addressed. Did you forget to commit your changes?

File: api-reference/beta/resources/searchalterationoptions.md [hunk 14]
```diff
@@ -19,7 +19,7 @@ Provides the search alteration options for spelling correction.
...
 |enableModification|Boolean|Indicates whether spelling modifications are enabled. If enabled, user will get the search results for corrected query **when there are no results** for the original query with typos and get the spelling modification information in **queryAlterationResponse** property of the [response](/graph/api/resources/searchresponse?view=graph-rest-beta&preserve-view=true). Optional.|
```
[the remark this answers is not in the document]
Comment (ZChristine) [reply api-reference/beta/resources/searchalterationoptions.md:22]: the change should be pushed to origin now. can you help check the change? @JarbasHorst

File: api-reference/v1.0/resources/searchrequest.md [hunk 15]
```diff
@@ -40,13 +42,18 @@ The following is a JSON representation of the resource.
...
-  "from": 1024,
-  "size": 1024,
   "fields": ["String"],
-  "enableTopResults": true  
+  "from": "Int32",
+  "query": {"@odata.type": "microsoft.graph.searchQuery"},
+  "queryAlterationOptions": {"@odata.type": "microsoft.graph.searchAlterationOptions"},
+  "resultTemplateOptions": [{"@odata.type": "microsoft.graph.resultTemplateOption"}],
+  "size": "Int32",
+  "aggregations": {"@odata.type": "microsoft.graph.sortProperty"}
```
Comment (Lauragra) [inline api-reference/v1.0/resources/searchrequest.md:56]: Please delete - this is a duplicate.

File: api-reference/v1.0/resources/searchrequest.md [hunk 16]
```diff
@@ -21,16 +21,18 @@ The JSON blob contains the types of resources expected in the response, the unde
...
 |from|Int32|Specifies the offset for the search results. Offset 0 returns the very first result. Optional.|
 |query|[searchQuery](searchquery.md)|Contains the query terms. Required.|
+|queryAlterationOptions|[searchAlterationOptions](searchalterationoptions.md)|Query alteration options formatted in a JSON blob that contains two optional flags related to spelling correction. Optional. |
+|resultTemplateOptions|[resultTemplateOption](resulttemplateoption.md) collection|Provides the search result template options to render search results from connectors.|
 |size|Int32|The size of the page to be retrieved. Optional.|
 |sortProperties|[sortProperty](sortProperty.md) collection|Contains the ordered collection of fields and direction to sort results. There can be at most 5 sort properties in the collection. Optional.|
```
Comment (Lauragra) [inline api-reference/v1.0/resources/searchrequest.md:37]: Are we missing the trimDuplicates property? I see that in the CSDL.

File: api-reference/v1.0/resources/enums.md [hunk 17]
```diff
@@ -1722,6 +1722,13 @@ Possible values for user account types (group membership), per Windows definitio
 |strategic|
 |unknownFutureValue|
 
+### searchAlterationType values
+
+| Member | Description |
+|:---------------|:----------|
+|modification|Get search results for the corrected alterate query and the spelling correction information for typos.|
```
Comment (Lauragra) [inline api-reference/v1.0/resources/enums.md:1729]: We don't include descriptions in the enums topic; just a single column table with the list of members. This is for API Doctor validation and not intended to be public facing. If the value names are not clear/intuitive, please define them in an H3 section in the alterationResponse topic.

File: api-reference/v1.0/resources/searchrequest.md (hunk 16 above)
Comment (ZChristine) [reply api-reference/v1.0/resources/searchrequest.md:37]: this is another feature. not included in the speller and MRT.

**Pull request merged**